kinto/views/permissions.py

```diff
@@ -82,6 +82,17 @@ def get_records(self, filters=None, sorting=None, pagination_rules=None,
         # Check settings for every allowed resources.
         from_settings = allowed_from_settings(self.request.registry.settings, principals)
 
+        # Add additional resources and permissions defined in settings/plugins
+        bucket_perms = from_settings.get('')
+        if bucket_perms:
+            for bucket_perm in bucket_perms:
```

Comment:
This should probably be renamed `root_perms` now.
And I guess it can be simplified:
```py
for root_perm in from_settings.get('', []):
    resource_name, _ = ...
```